### PR TITLE
Work on new TreeView replacement

### PR DESCRIPTION
**Pull request by flegall2014**
Provided replacement for CredentialsModel -> CredentialModel
Provided replacement for CredentialsModelFilter -> CredentialModelFilter
Provided replacement for CredentialsView -> CredentialView

### CODE REVIEW

**Comment by limpkin** [comment]
@flegall2014 es tu sur d'avoir tout pushé? la CI n'arrive pas à compiler.

File: gui.pro [hunk 1]
```diff
@@ -21,26 +21,30 @@ SOURCES += src/main_gui.cpp \
...
     src/FilesManagement.cpp \
-    src/SSHManagement.cpp
+    src/SSHManagement.cpp \
+    src/treeitem.cpp \
+    src/serviceitem.cpp \
+    src/loginitem.cpp \
+    src/rootitem.cpp \
+    src/credentialmodel.cpp \
+    src/credentialmodelfilter.cpp \
+    src/credentialview.cpp
```
Comment (raoulh) [inline gui.pro]: CamelCase filenames please

File: src/CredentialsManagement.h [hunk 2]
```diff
@@ -19,63 +19,147 @@
...
+    //! Credential updated
+    void onCredentialUpdated(const QString &service, const QString &login, const QString &description, bool success);
 
+    //! Save selected credential
+    void saveSelectedCredential(const QModelIndex &proxyIndex=QModelIndex());
+
+    //! Button EnterMMM clicked
     void on_pushButtonEnterMMM_clicked();
+
+    //! Button discard clicked
```
Comment (raoulh) [inline src/CredentialsManagement.h]: Why all those useless comments?
Please avoid that, it helps no one, it's juste noise.
https://testing.googleblog.com/2017/07/code-health-to-comment-or-not-to-comment.html

File: src/credentialview.cpp [hunk 3]
```diff
@@ -1,8 +1,72 @@
...
 
+ConditionalItemSelectionModel::ConditionalItemSelectionModel(TestFunction f, QAbstractItemModel *model)
+    : QItemSelectionModel(model)
+    , cb(f)
+    , lastRequestTime(0)
+{
+
+}
+
+//-------------------------------------------------------------------------------------------------
```
Comment (raoulh) [inline src/credentialview.cpp]: Don't add those useless comment lines. it helps no one

File: src/CredentialsManagement.h [hunk 4]
```diff
@@ -33,66 +36,129 @@ class CredentialsManagement : public QWidget
 {
     Q_OBJECT
 
-public:
+public:    
+    //-------------------------------------------------------------------------------------------------
+    // Constructor & destructor
+    //-------------------------------------------------------------------------------------------------
```
Comment (raoulh) [inline src/CredentialsManagement.h]: PLease... we all know what a constructor and a destructor is. there is no need to comment those...

File: src/credentialmodel.cpp [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+
+//-------------------------------------------------------------------------------------------------
+
+void CredentialModel::setClearTextPassword(const QString &sServiceName, const QString &sLoginName, const QString &sPassword)
+{
+    // Retrieve target service
+    ServiceItem *pTargetService = m_pRootItem->findServiceByName(sServiceName);
+
+    // Got a match
+    if (pTargetService != nullptr) {
```
Comment (raoulh) [inline src/credentialmodel.cpp]: Please keep the { on a new line to keep code consistency with existing code

**Comment by raoulh** [comment]
2 things here:
- Before I can test, the build should be fixed
- Before any merge can be done, the code/filenames formating should be fixed at least

File: src/CredentialsManagement.h (hunk 2 above)
Comment (limpkin) [inline src/CredentialsManagement.h]: you don't pay him by the hour ;)

**Pull request closed without merging**